server/warcwriter.go

```diff
@@ -59,11 +60,13 @@ func newWarcWriter(s settings.Settings, db database.DbAdapter, c *config.ConfigO
 		subCollection:    recordMeta.GetSubCollection(),
 		filePrefix:       createFilePrefix(c.GetMeta().GetName(), recordMeta.GetSubCollection(), now(), c.GetCollection().GetCollectionDedupPolicy()),
 	}
-	switch s.WarcVersion() {
+	switch warcVersion := s.WarcVersion(); warcVersion {
 	case gowarc.V1_1:
 		ww.revisitProfile = gowarc.ProfileIdenticalPayloadDigestV1_1
 	case gowarc.V1_0:
 		ww.revisitProfile = gowarc.ProfileIdenticalPayloadDigestV1_0
+	default:
+		panic(fmt.Sprintf("unhandled gowarc version: '%s'", warcVersion))
```

Comment:
```suggestion
		panic(fmt.Sprintf("unsupported WARC version: '%s'", warcVersion))
```

Are you confusing the version of the WARC spec and the version of the gowarc library?

Reply:
To add some context. It is already impossible to specify an invalid WARC version because of the accessor found here: https://github.com/nlnwa/veidemann-contentwriter/blob/ca66fa34fcfb7da4e115f1e7191e320cd07535a5/settings/settings.go#L58-L68
which enforce the flag specified here:
https://github.com/nlnwa/veidemann-contentwriter/blob/ca66fa34fcfb7da4e115f1e7191e320cd07535a5/main.go#L27

Reply:
> Are you confusing the version of the WARC spec and the version of the gowarc library?

I think I simply wrote the wrong thing. In my mind, I was always talking about the spec itself. ;) WIll fix

Reply:
> To add some context. It is already impossible to specify an invalid WARC version because of the accessor found here:

Say that someone adds a new specification, let's say `1.2`, what happens when you do so? First it will fail at line 66 in `settings.go` in the file that you linked to which is great! But then after the developer adds `1.2` to that switch so that we avoid the `default` case what happens?

The area of the code this PR touches will not fail, in fact, it will continue, potentially leading to weird bugs. This means that the developer must be responsible for finding all the places in the code that he/she needs to update, and cannot rely on the program itself to tell him/her.

What I am suggesting we do instead is to let the developer know that this switch needs updating, otherwise, we might end up in a bad state. One easy way of doing this is to `panic` in the default case unless we somehow can recover from the default case.

Reply:
You are of course right :+1: